libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -853,6 +853,7 @@ int lua_dirlist(lua_State *L) {
     int i;
     const char *path = luaL_checkstring(L, 1);
     
+    WITH_SEMAPHORE(AP::FS().get_semaphore(path));
     /* open directory */
     auto dir = AP::FS().opendir(path);
```

Comment:
`WITH_DIRHANDLE(....)`

that's a random idea to avoid the WITH_SEMAPHORE and opendir being intrisically linked (oir you get readdir failing randomly)

Reply:
Went with @tpwrules suggestion